### ISSUE TITLE
Unknown type integer error

### ISSUE THREAD

**Issue by Muffo** [mesonbuild/meson#3044]
I am using the following options taken directly from the [online documentation](http://mesonbuild.com/Build-options.html):

```
option('someoption', type : 'string', value : 'optval', description : 'An option')
option('other_one', type : 'boolean', value : false)
option('combo_opt', type : 'combo', choices : ['one', 'two', 'three'], value : 'three')
option('integer_opt', type : 'integer', min : 0, max : 5, value : 3)
option('free_array_opt', type : 'array', value : ['one', 'two'])
option('array_opt', type : 'array', choices : ['one', 'two', 'three'], value : ['one', 'two'])
```

When I build I get the following error:

```
[0/1] Regenerating build files.
The Meson build system
Version: 0.44.0
Source dir: /Users/Muffo/Devel/Repos/mesontest
Build dir: /Users/Muffo/Devel/Repos/mesontest/builddir
Build type: native build

Meson encountered an error in file meson_options.txt, line 4, column 0:
Unknown type integer.

A full log can be found at /Users/Muffo/Devel/Repos/mesontest/builddir/meson-logs/meson-log.txt
FAILED: build.ninja 
/usr/local/opt/python3/bin/python3.6 /usr/local/bin/meson --internal regenerate /Users/Muffo/Devel/Repos/mesontest /Users/Muffo/Devel/Repos/mesontest/builddir --backend ninja
ninja: error: rebuilding 'build.ninja': subcommand failed
```

I guess it's a typo in the documentation, however I am just starting to learn Meson and I don't know what should be the right syntax.

**Additional info:**

```
$ meson --version
0.44.0
```

**Comment by QuLogic** [comment]
It's a missing feature that was added in #3007, but is not in any release yet.

**Comment by nirbheek** [comment]
We need to fix the documentation to say that this feature is only available from 0.45 onwards.

**Comment by QuLogic** [comment]
Can be closed now.
